core/src/test/scala/org/apache/spark/util/collection/OpenHashSetSuite.scala

```diff
@@ -255,4 +255,16 @@ class OpenHashSetSuite extends SparkFunSuite with Matchers {
     val set = new OpenHashSet[Long](0)
     assert(set.size === 0)
   }
+
+  test("support for more than 12M items") {
+    val cnt = 12000000 // 12M
+    val set = new OpenHashSet[Int](cnt)
+    for (i <- 0 until cnt) {
+      set.add(i)
+
+      val pos1 = set.addWithoutResize(i) & OpenHashSet.POSITION_MASK
+      val pos2 = set.getPos(i)
+      assert(pos1 == pos2)
+    }
```

Comment:
nit: Is it better to add the following to check each value after adding all, too?
```
for (i <- 0 until cnt) {
  assert(set.contains(i))
}
```

Reply:
If we want to add the check, we can also add it inside the loop. Another loop seems unnecessary to me.

Reply:
My intention was to verify whether small values (e.g. `0, 1, 2, 3`) are still valid after several resizings.

Reply:
I see. I think that is not what this wants test and there should be other tests to make sure it works.

Reply:
The original test performed a check by using `map.iterator.count()`. But, this is not a strong preference.

Reply:
oh, the original test performed the count check to see how many values are invalid. They are invalid values because the index is wrong due to wrong position mask in OpenHashSet.

This rewritten test tests directly the index of OpenHashSet.